go/vt/mysqlctl/mysqld.go

```diff
@@ -978,6 +978,7 @@ func (mysqld *Mysqld) executeMysqlScript(connParams *mysql.ConnParams, sql io.Re
 // 'defer os.Remove()' statement.
 func (mysqld *Mysqld) defaultsExtraFile(connParams *mysql.ConnParams) (string, error) {
 	var contents string
+	connParams.Pass = strings.Replace(connParams.Pass, "#", "\#", -1)
```

Comment:
This should be `\\#`. Can you also write a test to show that it works correctly?

Reply:
Noted,  will address this shortly.